### ISSUE TITLE
[CI] Deprecations in `pytest==7.0.0` make tests fail

### ISSUE THREAD

**Issue by abravalheri** [pypa/setuptools#3079]
[Pytest 7](https://pypi.org/project/pytest/#history) that just came out a few hours ago is currently issuing a deprecation warning for `pytest-flake8`, which makes `setuptools` tests to fail:

```
$ tox
...
...
E   pytest.PytestDeprecationWarning: <class 'pytest_flake8.Flake8Item'> is not using a cooperative constructor and only takes {'fspath', 'parent'}.
E   See https://docs.pytest.org/en/stable/deprecations.html#constructors-of-custom-pytest-node-subclasses-should-take-kwargs for more details.
...
...
E   pytest.PytestRemovedIn8Warning: The (fspath: py.path.local) argument to Flake8Item is deprecated. Please use the (path: pathlib.Path) argument instead.
E   See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
```

This problem seems to be already reported in the project issue tracker: https://github.com/tholo/pytest-flake8/issues/83.

As a temporary workaround I believe we can take one of the following approaches:
1. Ignore those specific warnings in `pytest.ini`
2. Disable temporarily `pytest-flake8`
3. Add a version cap to the `pytest` test dependency.

Probably the first approach is the most straight forward?

**Comment by abravalheri** [comment]
Note, this issue also seems to apply to `pytest-perf`:

```
.tox/python/lib/python3.8/site-packages/pytest_perf/plugin.py:21: in pytest_collect_file
    return File.from_parent(parent, fspath=path)
...
E   pytest.PytestRemovedIn8Warning: The (fspath: py.path.local) argument to File is deprecated. Please use the (path: pathlib.Path) argument instead.
E   See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
```

**Comment by jaraco** [comment]
This issue probably affects every project based on jaraco/skeleton. Maybe it would be best to address it there.

**Comment by jaraco** [comment]
The issue also affects pytest-black and pytest-mypy, but those are still disabled on this project. It may also affect pytest-checkdocs.

**Comment by jaraco** [comment]
Reported the issue to https://github.com/jaraco/skeleton/issues/57.

**Comment by jaraco** [comment]
Merged as 8c0228f35.

**Comment by jaraco** [comment]
I'm addressing a follow-up issue in https://github.com/pytest-dev/pytest/discussions/9616.
